Fix: Enhance settings upsert logic to work with current enhancements of the settings API

warber opened the pull request:
Changes in upsert logic:

* get settings2.0 object with matching origin Object ID instead of external ID, if a setting obj was found and tenant version is < 1.262 we print the warning
* Deleted PUT operation on settings endpoint as it is not needed. POST endpoint now handles both, creating and updating settings objects

Further:
* deleted settings download filters that are not needed anymore with recent changes in settings fw
* Adjusted methods names of ConfigClient interface to contain reflect the fact that it is manipulating Configs as these interfaces are embedded into a bigger one and its clearer what each method does

btw: I don't know why legacy integration tests fail on GH. I ran them from my local machine, and it was running fine.

UnseenWizzard commented on pkg/client/client.go:
```diff
@@ -293,13 +293,13 @@ func isNewDynatraceTokenFormat(token string) bool {
 		// only consider objects with an external id that matches the expected one
 		Filter: func(o DownloadSettingsObject) bool {
-			return o.ExternalId == externalId
+			return o.ObjectId == obj.OriginObjectId
```
wait, so we're fulling letting the Settings framework handle the "upsert" logic? 

does making this call even make sense anymore then? If it's just used for the old version + existing object check maybe we can just implement a GET by object id rather than iterating?

warber replied:
yes, it's completely handled by the settings fw.
yes, actually a get would be also possible. didn't thought about that :+1:

UnseenWizzard replied:
As is we load and iterate over potentially hundreds of settings just to find a specific one. 
I would really suggest we replace it with just a get by objectId

warber replied:
just saw that the current implementation of `DynatraceClient` does not have a method to Get a single object. Are you OK with doing this "optimization" in a follow-up ? :)

warber commented:
@UnseenWizzard I've addressed your comment (GET single setting object instead of iterating) 
Addidionally, I've done a little cleanup --> renaming the methods of ConfigClient to have the name "config(s)" in it to make it more clear what they are targeting, since they are embedded into a larger interface with the other interfaces for settings and entities....

UnseenWizzard commented on pkg/client/client.go:
```diff
@@ -291,98 +297,57 @@ func isNewDynatraceTokenFormat(token string) bool {
-		if err != nil {
-			return DynatraceEntity{}, fmt.Errorf("failed to build settings object for upsert: %w", err)
+		if fetchedSettingObj != nil {
+			log.Warn("Unable to update Settings 2.0 object of schema %q and object id %q on Dynatrace tenant with a version < 1.262.0", obj.SchemaId, obj.OriginObjectId)
```
Only just noticed 'Dynatrace environment' rather than tenant please.

UnseenWizzard commented on pkg/client/client.go:
```diff
@@ -291,98 +297,57 @@ func isNewDynatraceTokenFormat(token string) bool {
-			return DynatraceEntity{}, fmt.Errorf("failed to parse response: %w", err)
-		}
+	if !success(resp) {
+		return DynatraceEntity{}, fmt.Errorf("failed to update settings object with externalId %s (HTTP %d)!\n\tResponse was: %s", externalId, resp.StatusCode, string(resp.Body))
```
`failed to upsert` rather than "update"

UnseenWizzard reviewed (commented):
looks good to me, comments are just about wording in user facing messages
